### ISSUE TITLE
Embedding loss is exactly 0 at a per-device batch of 1, and the `embedding` template's `batch_size: auto` picks 1 on CPU, MPS and GPUs under ~45 GiB

### ISSUE THREAD

**Issue by MakazhanAlpamys** [MakazhanAlpamys/Soup#1234]
## What happens

`task: embedding` with `embedding_loss: contrastive` uses in-batch negatives: row *i*'s positive is the target, and the other rows' positives are the negatives. With one row per device batch there are no negatives. The similarity matrix is 1x1, and the cross-entropy is `x - logsumexp([x]) = 0` exactly, with a gradient of exactly 0.

The shipped template (`src/soup_cli/templates/embedding.yaml`: `BAAI/bge-base-en-v1.5`, `batch_size: auto`, `quantization: none`, `lora.r: 64`, `max_length: 512`, `embedding_loss: contrastive`) resolves `auto` to 1 in most setups:

```python
from soup_cli.utils.gpu import estimate_batch_size, model_size_from_name

size_b = model_size_from_name("BAAI/bge-base-en-v1.5")
print(f"model_size_from_name('BAAI/bge-base-en-v1.5') = {size_b}")
for label, gib in [("CPU / MPS (memory_total_bytes=0)", 0), ("16 GiB", 16), ("24 GiB", 24),
                   ("40 GiB", 40), ("48 GiB", 48), ("80 GiB", 80)]:
    est = estimate_batch_size(model_params_b=size_b, seq_length=512,
                              gpu_memory_bytes=int(gib * 1024**3), quantization="none", lora_r=64)
    print(f"  {label:34s} estimate={est:2d} -> embedding.py:118 max(1, est // 3) = {max(1, est // 3)}")
```

```
model_size_from_name('BAAI/bge-base-en-v1.5') = 7.0
  CPU / MPS (memory_total_bytes=0)   estimate= 1 -> embedding.py:118 max(1, est // 3) = 1
  16 GiB                             estimate= 1 -> embedding.py:118 max(1, est // 3) = 1
  24 GiB                             estimate= 2 -> embedding.py:118 max(1, est // 3) = 1
  40 GiB                             estimate= 4 -> embedding.py:118 max(1, est // 3) = 1
  48 GiB                             estimate= 8 -> embedding.py:118 max(1, est // 3) = 2
  80 GiB                             estimate=16 -> embedding.py:118 max(1, est // 3) = 5
```

(`bge-base` is about 0.11B parameters. The name has no size marker, so the estimator falls back to 7B.)

End to end, the shipped template's own `training` block ran through the real `EmbeddingTrainerWrapper.setup()` and `trainer.train()` on CPU. Only the base was swapped, for a tiny random-init BERT built from a config (no download). The run was one epoch over 4 rows, with `gradient_accumulation_steps: 4` as in the template. A spy around `_compute_embedding_loss` recorded each call:

```
Auto batch size (embedding): 1
[template_auto_cpu] per_device_train_batch_size=1 GA=4
[template_auto_cpu] compute_loss calls (rows, loss): [(1, 0.0), (1, 0.0), (1, 0.0), (1, 0.0)]
[template_auto_cpu] logged (loss, grad_norm): [(0.0, 0.0)]
[template_auto_cpu] lora_B tensors changed: 0/4  max|lora_B| after: 0.000e+00
```

The same run with `get_gpu_info()` reporting a 24 GiB card (everything else the real code path), then the two controls:

```
[template_auto_24gib_simulated] per_device_train_batch_size=1 GA=4
[template_auto_24gib_simulated] compute_loss calls (rows, loss): [(1, 0.0), (1, 0.0), (1, 0.0), (1, 0.0)]
[template_auto_24gib_simulated] logged (loss, grad_norm): [(0.0, 0.0)]
[template_auto_24gib_simulated] lora_B tensors changed: 0/4  max|lora_B| after: 0.000e+00
[template_auto_48gib_simulated] per_device_train_batch_size=2 GA=4
[template_auto_48gib_simulated] compute_loss calls (rows, loss): [(2, 0.045558325946331024), (2, 0.015166403725743294)]
[template_auto_48gib_simulated] lora_B tensors changed: 4/4  max|lora_B| after: 2.000e-05
[template_explicit_2_control] per_device_train_batch_size=2 GA=1
[template_explicit_2_control] lora_B tensors changed: 4/4  max|lora_B| after: 3.001e-05
```

LoRA B starts at zero and never moves, so the saved adapter's delta is exactly zero. The run completes, logs `loss 0`, and nothing warns. An explicit `batch_size: 1` behaves the same way (4 micro-batches, all `0.0`), and gradient accumulation does not help because it adds no in-batch negatives.

## Why

`src/soup_cli/trainer/embedding.py:575-579` computes `similarity = anchor_emb @ pos_emb.T / temperature` with `labels = arange(batch)`. At batch 1 that is `cross_entropy([[x]], [0]) = 0`, and the gradient `1 - softmax([x]) = 0`. `embedding.py:117-118` divides the estimate by 3, and on the template the estimate is at most 4 below ~45 GiB for two reasons: `utils/gpu.py:280-281` returns 1 when `memory_total_bytes == 0` (CPU and MPS), and `utils/gpu.py:373` guesses 7B for any name without a size marker, which is ~60x too large for `bge-base`. The triplet path falls back to contrastive when the data has no `negative` column (`embedding.py:145-150`), so it inherits the same zero.

## Fix path

Make one-row contrastive batches impossible rather than silent. Floor the auto batch at 2 whenever the effective loss is contrastive, including the triplet-to-contrastive fallback. Refuse an explicit `batch_size: 1` with `embedding_loss: contrastive` at config parse, with a message that names the loss: "contrastive in-batch negatives need batch_size >= 2; use triplet (with negatives) or cosine for batch 1". Add a runtime guard in `_compute_embedding_loss` that raises if `similarity.size(0) < 2`. The last partial batch of an epoch (for example 5 rows at batch 2) also produces a one-row batch, so either set `dataloader_drop_last` for contrastive or let the guard skip that batch explicitly.

Separately, and lower priority: the 7B default in `model_size_from_name` is what drives the estimate this low for BERT-family embedding models. Reading the parameter count from the Hub config, or recognising `small` / `base` / `large` for encoder models, would stop the estimator from under-sizing every embedding run. That alone does not fix the defect, because CPU/MPS still resolve to 1.

## Acceptance

- [ ] A CPU test that runs `EmbeddingTrainerWrapper` on a tiny BERT with the shipped template's `training` block. On current code it resolves batch 1, logs loss 0, and `lora_B` stays all zeros. After the fix it either trains (batch >= 2, `lora_B` changes) or refuses with the message above.
- [ ] Parse-time refusal for `batch_size: 1` + `embedding_loss: contrastive`, asserting the message names both the loss and the batch size.
- [ ] Variants covered: `batch_size: auto` with `memory_total_bytes` 0 (CPU/MPS), 16, 24 and 40 GiB; explicit `batch_size: 1`; `batch_size: 1` with `gradient_accumulation_steps > 1`; a last partial batch of one row; `embedding_loss: triplet` with no `negative` column (the fallback path).
- [ ] `triplet` with negatives and `cosine` at batch 1 keep working (they do not use in-batch negatives).
